### PR TITLE
chore(chart): add custom legend examples

### PR DESCRIPTION
**Pull request by dlabrecq**
A couple teams recently asked about creating custom legends. It would be helpful to add the following to our legend examples for future reference.

1. Show how to add links in a custom legend
2. Show a legend with custom label layout and styles

Fixes https://github.com/patternfly/patternfly-react/issues/4272

chrome-capture

### CODE REVIEW

File: packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md [hunk 1]
```diff
@@ -480,6 +421,234 @@ class InteractiveLegendChart extends React.Component {
 }
 ```
 
+```js title=Legend-tooltips
+import React from 'react';
+import { ChartLabel, ChartLegend, ChartPie, ChartThemeColor } from '@patternfly/react-charts';
```
Comment (redallen) [inline packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:426]: Add imports here.

Comment (dlabrecq) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:426]: Ok, I assumed lint:md was run locally, as part of the build target? 

Fixed

Comment (dlabrecq) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:426]: FYI, I intentionally added an error, but neither the `build` and `lint` targets pick it up. I have to run `lint:md` separately.

Comment (redallen) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:426]: Yeah, `lint:md` isn't a part of `lint`. We could add it to be now that `lint` doesn't take hours :)

Comment (ddonahue007) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:426]: I should have waited to approve until the tests ran through fully, that's my bad... :(

**Pull request merged**